driver-core/src/main/com/mongodb/internal/connection/OidcAuthenticator.java

```diff
@@ -69,11 +73,20 @@
  */
 public final class OidcAuthenticator extends SaslAuthenticator {
 
-    private static final List<String> SUPPORTED_PROVIDERS = Arrays.asList("aws");
+    private static final String TEST_ENVIRONMENT = "test";
```

Comment:
This is only used (enabled) in unified tests. The "Entities" class could instead check if this environment is set, and supply the test callback under OIDC_CALLBACK_KEY, which is a mild abuse of mechanism properties, but would avoid the (limited) code needed in the driver to support this.